x/auctions/keeper/hooks.go

```diff
@@ -70,15 +74,7 @@ func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, _ int64)
 	if err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.AuctionCollectorName, types.ModuleName, accumulatedCoins); err != nil {
 		return
 	}
-}
-
-// isValidBid checks if the bid is valid
-func isValidBid(lastBid types.Bid) bool {
-	_, err := sdk.AccAddressFromBech32(lastBid.Sender)
-	if lastBid.Amount.Amount.IsPositive() && lastBid.Sender != "" && err == nil {
-		return true
-	}
-	return false
+	writeFn()
```

Comment:
moved up, within the if statement
```suggestion
```

Reply:
Yeah, as Rama said the entire function should be atomic because if we have an error above, and move funds from auction collector, the bidder will have more funds in the pot.

Reply:
I think this would be nicer with a `defer func() {...}`